### PR TITLE
Dynamic simulation output variables supplier

### PR DESCRIPTION
**Pull request by Lisrte**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

Dynamic simulation `CurveMapping` handle curve for dynamic models.

**What is the new behavior (if this is a feature change)?**
`CurveMapping` is replaced with `OutputVariableMapping` handling curves and final state values for equipment with or without dynamic models.
Final state values results are provided by `SimulationResult.final_state_values`.

**Does this PR introduce a breaking change or deprecate an API?**
- [x] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [x] The *Breaking Change* or *Deprecated* label has been added
- [x] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

Replace `CurveMapping::add_curve` and `CurveMapping::add_curve` with `OutputVariableMapping::add_dynamic_model_curves`.

**Other information**:

Linked to the powsybl-dynawo PR [413](https://github.com/powsybl/powsybl-dynawo/pull/413)
Without the PR, mapping with more than two curves will end with `pandas.errors.InvalidIndexError`.

### CODE REVIEW

File: pypowsybl/dynamic/impl/simulation_result.py [hunk 1]
```diff
@@ -37,4 +38,8 @@ def _get_all_curves(self) -> pd.DataFrame:
         curve_name_lst = _pp.get_all_dynamic_curves_ids(self._handle)
         df_curves = [self._get_curve(curve_name)
                      for curve_name in curve_name_lst]
-        return pd.concat(df_curves, axis=1) if df_curves else pd.DataFrame()
+        return pd.concat(df_curves, axis=1).ffill() if df_curves else pd.DataFrame()
```
Comment (HugoKulesza) [inline pypowsybl/dynamic/impl/simulation_result.py:41]: By curiosity, why is .ffill needed here ?

File: pypowsybl/dynamic/impl/output_variable_mapping.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2023, RTE (http://www.rte-france.com)
```
Comment (HugoKulesza) [inline pypowsybl/dynamic/impl/output_variable_mapping.py]: Just a wrong date here

File: java/src/main/java/com/powsybl/dataframe/dynamic/OutputVariablesSeries.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2020-2022, RTE (http://www.rte-france.com)
```
Comment (HugoKulesza) [inline java/src/main/java/com/powsybl/dataframe/dynamic/OutputVariablesSeries.java]: Same here

File: integration_tests/test_dynawo.py [hunk 4]
```diff
@@ -9,13 +9,16 @@
...
+import logging
 
 def test_simulation():
     """
     Running that test requires to have installed dynawo,
     and configured its path in your config.yml.
     """
+
+    logging.basicConfig()
+    logging.getLogger('powsybl').setLevel(1)
```
Comment (HugoKulesza) [inline integration_tests/test_dynawo.py]: This is probably not useful to keep for future run of integration tests

File: pypowsybl/dynamic/impl/simulation_result.py (hunk 1 above)
Comment (Lisrte) [reply pypowsybl/dynamic/impl/simulation_result.py:41]: Curves are represented as a map of `DoubleTimeSeries` in Powsybl. When creating a `DoubleTimeSeries` for an individual curve, entries with the same value as the previous one are skipped. Thus when we concat  curves series we have to fill forward in order to get these values back.

**Review by HugoKulesza — APPROVED** [review]

**Pull request merged**